### DISCUSSION TITLE
Stuw niet gebruiken als benedenstroomse waterstand te hoog is

### DISCUSSION THREAD

**Discussion by tessa-andringa-sweco** [Deltares/Ribasim#501] [Q&A] [answered]
Hoe kunnen we in het Ribasim model aangeven dat er alleen water via een stuw mag stromen als de waterstand benedenstrooms niet te hoog is? 
Als voorbeeld; de benedenrandvoorwaarde in ons gebied is de IJssel waar het water via een stuw naar toe stroomt. Dat water stroomt alleen via die stuw als de IJssel niet te hoog staat (anders wordt de stuw afgesloten en gaat het via een pomp naar de IJssel)

**Comment by SouthEndMusic** [comment, marked as answer]
Hallo @tessa-andringa-sweco,

Dit klinkt als een toepassing van de `DiscreteControl` node, zie [hier](https://deltares.github.io/Ribasim/core/usage.html#discetecontrol). Dan kan je de logica implementeren als volgt:
- In de `condition` geef je aan: de node ID van de `DiscreteControl` node, de node ID van het IJssel bassin, de variabele "level" en de grenswaarde van het level;
- In `logic` geef je aan welke control state geldt als de level conditie wel waar is ('T', i.e. de level is hoger dan je grenswaarde) of niet waar is ('F', i.e. de level is lager dan je grenswaarde). Deze control states zijn een naam, die een betekenis krijgen in de `static` tabel van verbonden nodes;
- In de graaf van je model, maak control edges van je `DiscreteControl` node naar de nodes die gecontroleerd moeten worden, i.e. de `Pump` en `TabulatedRatingCurve`;
- In de static tabel van deze nodes, definieer voor de hierboven genoemde control states wat die nodes voor parameters moeten hebben als ze in een bepaalde control state zijn. Als je een node geheel wil uitschakelen kan je de variabele 'active' op false zetten.

Ik hoop dat dit een antwoord geeft op je vraag, ik hoor het graag.

Mvg,

Bart

**Comment by SouthEndMusic** [comment]
Ik kom er net achter dat een `TabulatedRatingCurve` (de)activeren met control pas werkt na mergen van deze PR: https://github.com/Deltares/Ribasim/pull/446.
